notebooks/Notebook3.md

```diff
@@ -18,17 +26,46 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+This week I "completed" a draft program in my DSL, the card game Hearts, when I realized that there was 
+something severely wrong with my entire design. I had completely forgotten that Scala is not a 
+dynamically-typed language, and thus, I couldn't use classes in place of their subclasses, like I had in my 
+Python implementation of the language. The majority of the week was spent testing out different methods for 
+making my existing design work: the Dynamic trait, generic types, and even switching host languages. At the 
+very end of my work this week, I think I found an acceptable solution, which involves a HashMap of variables 
+and constants, plus liberal use of the class extension functionality of Scala, but I definitely want to talk 
+to other people in class about other ways I might go about tackling the problem. It definitely was 
+disappointing to go from feeling like my language was at a really good place to having such a severe setback, 
+but perhaps it will be for the best; the way I'm planning on doing things now won't require the user to have 
+to understand OOP, which would be really nice.
+
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+Honestly, I just have to get these type issues ironed out and I should have something that, while ugly, 
+actually does work the way I intend it to.
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+I just want to run the various options I discussed above by my partners and talk about the pros and cons. I'll 
+also definitely need to chat with Prof Ben about if there are any Scala features I might be missing or other 
+ways to tackle the problem that I haven't thought of yet.
+
 **How much time did you spend on the project this week? If you're working in a
 team, how did you share the work?**
 
+I spent a significant amount of time on the project this week, probably close to 8 or 9 hours, not including 
+class time and this reflection.
```

Comment:
Don't overwhelm yourself. Remind yourself that this project will take a lot of time, and its more important to make progress then to finish.